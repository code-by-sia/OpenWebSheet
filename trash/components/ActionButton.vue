<template>
    <button class="action" :disabled="disabled" @click="$emit('click')" data-qa="action">
        <slot>
            <slot name="prefix"/>
            {{label}}
            <slot name="postfix"/>
        </slot>
    </button>
</template>

<script lang="ts">
import { Component, Prop } from 'vue-property-decorator';
import Vue from 'vue';

@Component({name: 'ActionButton'})
export default class ActionButton extends Vue {
  @Prop()
  public label?: string;

  @Prop({default: false})
  public disabled!: boolean;
}
</script>

<style lang="scss" scoped>
    button.action {
        border: solid thin #ccc;
        font-size: x-small;
        border-radius: 2px;
        background: linear-gradient(to top, #ece9e6, #ffffff);
        padding: 3px;
        width: 100%;

        &.on {
            background: linear-gradient(to top, #cccccc, #efefef);
        }
    }
</style>
