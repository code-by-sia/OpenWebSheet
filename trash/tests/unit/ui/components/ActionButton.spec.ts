import { shallowMount } from '@vue/test-utils';
import ActionButton from '@/components/ActionButton.vue';

describe('ActionButton', () => {
  it('is a vue instance', () => {
    const wrapper = shallowMount(ActionButton, {});
    expect(wrapper.isVueInstance()).toBe(true);
  });

  it('should renders empty button by default', () => {
    const wrapper = shallowMount(ActionButton, {});
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders label', () => {
    const wrapper = shallowMount(ActionButton, {
      propsData: {
        label: 'LABEL',
      },
    });
    expect(wrapper.find('[data-qa=action]').text()).toBe('LABEL');
    expect(wrapper.find('[data-qa=action]').attributes('disabled')).toBeFalsy();
  });

  it('should reflect disabled on the target button', () => {
    const wrapper = shallowMount(ActionButton, {
      propsData: {
        label: 'LABEL',
        disabled: true,
      },
    });
    expect(wrapper.find('[data-qa=action]').attributes('disabled')).toBeTruthy();
  });


  it('should uses default slot when it exists', () => {
    const wrapper = shallowMount(ActionButton, {
      slots: {
        default: '<strong data-qa="custom-slot">CUSTOM</strong>',
      },
    });

    expect(wrapper.find('[data-qa=custom-slot]').exists()).toBe(true);
    expect(wrapper.find('[data-qa=custom-slot]').text()).toBe('CUSTOM');
  });

  it('should append prefix and postfix slots', () => {
    const wrapper = shallowMount(ActionButton, {
      propsData: {
        label: 'LABEL',
      },
      slots: {
        prefix: '<strong data-qa="pre">[PRE]</strong>',
        postfix: '<strong data-qa="post">[POST]</strong>',
      },
    });

    expect(wrapper.find('[data-qa=pre]').exists()).toBe(true);
    expect(wrapper.find('[data-qa=post]').exists()).toBe(true);
    expect(wrapper.find('[data-qa=action]').text()).toMatchSnapshot();
  });

});
