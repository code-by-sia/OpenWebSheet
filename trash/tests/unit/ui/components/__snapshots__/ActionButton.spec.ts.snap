// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButton should append prefix and postfix slots 1`] = `
"[PRE]
        LABEL
        [POST]"
`;

exports[`ActionButton should renders empty button by default 1`] = `
<button data-qa="action" class="action">

</button>
`;
